src/test/kotlin/com/terraformation/backend/db/DatabaseBackedTest.kt

```diff
@@ -3047,6 +3051,128 @@ abstract class DatabaseBackedTest {
     return row.id!!.also { inserted.variableWorkflowHistoryIds.add(it) }
   }
 
+  /**
+   * Asserts that a table (or a filtered subset of it) contains an expected set of records and
+   * nothing else.
+   *
+   * @param table The table whose contents should be examined.
+   * @param expected The set of records the table should contain.
+   * @param message Assertion failure message; defaults to the table name.
+   * @param where Optional query condition to assert on a subset of a table's contents. Only rows
+   *   matching the condition will be considered.
+   * @param clearPrimaryKeys If true, the primary key field(s) of the records from the database will
+   *   be cleared before comparing against [expected]. This can be used to ignore database-generated
+   *   IDs.
+   * @param transform Function to apply to records from the database before comparing them to
+   *   [expected]. Can be used to clear or hardwire specific fields that aren't relevant to the
+   *   behavior being tested.
+   */
+  protected fun <R : UpdatableRecord<R>> assertTableEquals(
```

Comment:
Callers should rarely need to use this variant of `assertTableEquals`. The variants below this require fewer arguments.